File: Python/programmers/etc/level2/카펫.py
def solution(brown, yellow):
    tot = brown+yellow
    for i in range(1, int(tot**0.5)+1):
        a, b = i, tot//i
        if tot%i == 0 and 2*(a+b-2) == brown:
            return [b, a]
        

brown, yellow = 10, 2
print(solution(brown, yellow)) # [4, 3]

brown, yellow = 8, 1
print(solution(brown, yellow)) # [3, 3]

brown, yellow = 24, 24
print(solution(brown, yellow)) # [8, 6]

File: Java/programmers/etc/level2/카펫.java
package CodeTest.Java.programmers.etc.level2;

public class 카펫 {
    public int[] solution(int brown, int yellow) {
        int total = brown+yellow;
        int a = 0, b = 0;

        for (int i=1; i<Math.sqrt(total)+1; i++) {
            a = i; b = total/i;
            if (total%i == 0 && 2*(a+b-2) == brown) {
                break;
            }
        }
        
        return new int[]{b, a};
    }
}
